propolis/src/inventory.rs

```diff
@@ -457,12 +457,17 @@ pub trait Entity: Send + Sync + 'static {
         &self,
         next: State,
         target: Option<State>,
+        phase: TransitionPhase,
         ctx: &DispCtx,
     ) {
     }
     /// Function dedicated to `State::Reset` event delivery so implementers do
     /// not need to create a more verbose `state_transition` implementation for
     /// emulating device reset.
+    ///
+    /// This is called as part of `TransitionPhase::Pre`.  Entities which
+    /// require logic in the `Post` phase should do so via the
```

Comment:
Would it be worth also splitting out the post phase as well? Letting devices implement `pre_reset` and/or `post_reset` as needed.

Reply:
Since `Entity::reset` is there for convenience, and few things (so far) need the explicit post actions, I'm inclined to just let them handle it via `Entity:state_transition`.